pkg/scheduler/scheduler.go

```diff
@@ -605,11 +605,25 @@ func (sched *Scheduler) scheduleOne() {
 			fwk.RunUnreservePlugins(pluginContext, assumedPod, scheduleResult.SuggestedHost)
 			sched.recordSchedulingFailure(assumedPod, err, SchedulerError, fmt.Sprintf("Binding rejected: %v", err))
 		} else {
-			klog.V(2).Infof("pod %v/%v is bound successfully on node %v, %d nodes evaluated, %d nodes were found feasible", assumedPod.Namespace, assumedPod.Name, scheduleResult.SuggestedHost, scheduleResult.EvaluatedNodes, scheduleResult.FeasibleNodes)
+			node, _ := sched.Cache().GetNodeInfo(scheduleResult.SuggestedHost)
+			klog.V(2).Infof("pod %v/%v is bound successfully on node %q, %d nodes evaluated, %d nodes were found feasible. Bound node resource: %q.", assumedPod.Namespace, assumedPod.Name, scheduleResult.SuggestedHost, scheduleResult.EvaluatedNodes, scheduleResult.FeasibleNodes, nodeResourceString(node))
```

Comment:
This change worries me bit about performance impact, but since this is run inside the bind go routine and is not in the main thread of the scheduler, I think it shouldn't have a noticeable performance impact. It would be good if you could run our benchmarks (with V=2) to check the impact.

Reply:
I ran the benchmark tests `go test ./test/integration/scheduler_perf/ -bench=. -args -v=2` and here are the results. It seems that the overhead is negligible. The overhead of 6.91% for `BenchmarkSchedulingMigratedInTreePVs/500Nodes/250Pods-12` is more like an outlier datapoint.


Test | Without Logging Change (ns/op) - Run1 | Without Logging Change (ns/op) - Run2 | Without Logging Change (ns/op) - Avg | With Logging Change (ns/op) - Run1 | With Logging Change (ns/op) - Run2 | With Logging Change (ns/op) - Avg | Dif
-- | -- | -- | -- | -- | -- | -- | --
BenchmarkScheduling/1000Nodes/0Pods-12 | 38889773 | 39859933 | 39374853 | 40475404 | 40930320 | 40702862 | 3.37%
BenchmarkScheduling/1000Nodes/1000Pods-12 | 40187244 | 40898029 | 40542636.5 | 40256567 | 40454137 | 40355352 | -0.46%
BenchmarkScheduling/5000Nodes/1000Pods-12 | 25581820 | 25097632 | 25339726 | 25517080 | 25086796 | 25301938 | -0.15%
BenchmarkSchedulingPodAntiAffinity/500Nodes/250Pods-12 | 9607668 | 9597078 | 9602373 | 10026349 | 9471098 | 9748723.5 | 1.52%
BenchmarkSchedulingPodAntiAffinity/500Nodes/5000Pods-12 | 10718499 | 11390427 | 11054463 | 11168541 | 11175246 | 11171893.5 | 1.06%
BenchmarkSchedulingPodAntiAffinity/1000Nodes/1000Pods-12 | 10349515 | 10444192 | 10396853.5 | 10229299 | 10152229 | 10190764 | -1.98%
BenchmarkSchedulingPodAntiAffinity/5000Nodes/1000Pods-12 | 28767295 | 28775522 | 28771408.5 | 28933773 | 28670505 | 28802139 | 0.11%
BenchmarkSchedulingSecrets/500Nodes/250Pods-12 | 9270177 | 9355098 | 9312637.5 | 9327944 | 9286557 | 9307250.5 | -0.06%
BenchmarkSchedulingSecrets/500Nodes/5000Pods-12 | 9122269 | 9246834 | 9184551.5 | 9079228 | 9378848 | 9229038 | 0.48%
BenchmarkSchedulingSecrets/1000Nodes/1000Pods-12 | 9528526 | 9589277 | 9558901.5 | 9656471 | 9529816 | 9593143.5 | 0.36%
BenchmarkSchedulingSecrets/5000Nodes/1000Pods-12 | 25957904 | 25948559 | 25953231.5 | 25782664 | 25881367 | 25832015.5 | -0.47%
BenchmarkSchedulingInTreePVs/500Nodes/250Pods-12 | 12939606 | 12520737 | 12730171.5 | 12849149 | 11440859 | 12145004 | -4.60%
BenchmarkSchedulingInTreePVs/500Nodes/5000Pods-12 | 12433723 | 12403172 | 12418447.5 | 12049491 | 12346644 | 12198067.5 | -1.77%
BenchmarkSchedulingInTreePVs/1000Nodes/1000Pods-12 | 17201093 | 17852084 | 17526588.5 | 17590277 | 17271761 | 17431019 | -0.55%
BenchmarkSchedulingInTreePVs/5000Nodes/1000Pods-12 | 68903470 | 67837075 | 68370272.5 | 66168349 | 67535921 | 66852135 | -2.22%
BenchmarkSchedulingMigratedInTreePVs/500Nodes/250Pods-12 | 10805065 | 13078367 | 11941716 | 12808091 | 12725404 | 12766747.5 | 6.91%
BenchmarkSchedulingMigratedInTreePVs/500Nodes/5000Pods-12 | 12738517 | 12444537 | 12591527 | 12965807 | 12544700 | 12755253.5 | 1.30%
BenchmarkSchedulingMigratedInTreePVs/1000Nodes/1000Pods-12 | 17556447 | 17383449 | 17469948 | 17544710 | 17609127 | 17576918.5 | 0.61%

Reply:
Shouldn't this amount detail have a higher verbosity level?

Reply:
I ran tests with v=2 (as @bsalamat suggested), and v=2 is the level of logs that this change may affect. Why do we need a higher level?

Reply:
It is a matter of semantics. I'm not sure what the different log levels in k8s are. But if other reviewers are ok with this, it should be fine. Btw, this should be implemented like so:

```
if klog.V(2) {
  getnodeinfo
  log
}
```

That way the query to nodeinfo and the string calculation is skipped when we are not logging.

Reply:
SG.